docs/LoaderDriverInterface.md

```diff
@@ -799,6 +787,31 @@ However, the driver should continue to support the command query via
 `vk_icdGetPhysicalDeviceProcAddr`, until at least a Vulkan version bump, because
 an older loader may still be attempting to use the commands.
 
+### Reason for adding `vk_icdGetPhysicalDeviceProcAddr`
+
```

Comment:
Makes sense to have this down here after the reader has had a chance to learn a bit more about the topic.